### DISCUSSION TITLE
Custom RequestValidationError but Swagger not update the schema

### DISCUSSION THREAD

**Discussion by bfss** [fastapi/fastapi#6281] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request, exc):
    errors = exc.errors()
    custom_error = []
    for error in errors:
        custom_error.append(error['msg'])
    return JSONResponse(status_code=422, content=jsonable_encoder({'detail': custom_error}))
```

### Description

After I custom the RequestValidationError above, the swagger's schema is still the original one.

### Operating System

Windows

### FastAPI Version

0.79.0

### Python Version

3.9.13

**Comment by iudeen** [comment]
Not sure if adding it this way changes OpenAPI doc.

**Comment by JarroVGIT** [comment]
@iudeen  is correct, adding a handler does not automagically add that to the generated `openapi.json`. You would have to add those to the endpoints themselves in order for them to show up in the spec, otherwise it will use the default definition: 

```python
validation_error_definition = {
    "title": "ValidationError",
    "type": "object",
    "properties": {
        "loc": {
            "title": "Location",
            "type": "array",
            "items": {"anyOf": [{"type": "string"}, {"type": "integer"}]},
        },
        "msg": {"title": "Message", "type": "string"},
        "type": {"title": "Error Type", "type": "string"},
    },
    "required": ["loc", "msg", "type"],
}

validation_error_response_definition = {
    "title": "HTTPValidationError",
    "type": "object",
    "properties": {
        "detail": {
            "title": "Detail",
            "type": "array",
            "items": {"$ref": REF_PREFIX + "ValidationError"},
        }
    },
}
```

**Comment by bfss** [comment]
@JarroVGIT Thanks, but how could I add them to endpoints?

**Comment by JarroVGIT** [comment, marked as answer]
It’s described [here in the docs](https://fastapi.tiangolo.com/advanced/additional-responses/) . You can add a 422 response in the `responses` parameter of the **decorator** and that will override the default.

**Comment by bfss** [comment]
@JarroVGIT Thanks again! I'll close this issue~
